src/lambda_toolshed/papillon/async.cljc

```diff
@@ -1,16 +1,13 @@
-(ns lambda-toolshed.papillon.async
-  #?(:cljs
-     (:require [clojure.core.async :as core.async]
-               [clojure.core.async.impl.protocols :refer [ReadPort]]
-               [cljs.core.async.interop :as core.async.interop :refer [p->c]])))
+(ns lambda-toolshed.papillon.async)
+
+(defprotocol Chrysalis
+  (eclose [this handler]))
```

Comment:
> eclose
[intransitive verb](https://www.merriam-webster.com/dictionary/intransitive)
ēˈklōz
-ed/-ing/-s
of an insect
: to emerge from the eggshell or pupal case

Reply:
linguistically it works, though does have me wondering if sticking too close to the entomological theme within the library, might cause/increase friction for non-native English speakers.

or cross pollination with finance industry terms like "e-close"

Reply:
"or cross pollination with ..." 
I see what you did there.

Reply:
But isn't it the Clojure way to find an underutilized word, give a definition, and get it into more mainstream usage in the Clojure community? 😜

Reply:
yeah, a lot of lib/repos do this same thing. To be clear, Chrysalis is probably fine, my concern is more around `eclose` and how it might lead to human parsing errors. On a quick read can be read as `close` instead, leading them to thinking it's about closing or dismissing a thing, instead of realizing or completion of a thing.

Reply:
I am okay with a different/better name.  I am even open to renaming the protocol.

I went with these to avoid conceptual collisions with other libraries' ideas of having an "Async Context" to let it stand independently.

I wanted to ensure they were distinct enough that someone didn't confuse this protocol with something else. 😉

Reply:
Any objections to `emerge`?

Reply:
`emerge` can work, it's a clear enough term that it'd be a lot less likely to cause language parsing errors.

if `emerge` doesn't quite click, the synonyms `surface` or `materialize` might be options.

yeah, finding the best term here is tough, when time allows I've been digging through CS literature to try and find the best answer (most things just use "async"). the problem of, how do we describe the context of asynchronous operations in a unique but also still related way.

something like `allochronic` (https://www.merriam-webster.com/dictionary/allochronic) might work in place of `async`, though from the definition is more about geologic time. possible that `allochronic` it's esoteric enough in this context that it would also cause confusion.

I'll keep looking just in case.

Reply:
`AsyncSystem` might also work for the protocol
https://en.wikipedia.org/wiki/Asynchronous_system